org-formation/650-identity-providers/microsoft-defender-cloud-roles.yaml

```diff
@@ -0,0 +1,1436 @@
+    Type: String
+    Description: Provide a role ARN name DefenderForCloud-AgentlessScanner for Servers
+      agentless scanning offering
+    AllowedPattern: '[-_a-zA-Z0-9]+'
+    Default: DefenderForCloud-AgentlessScanner-stackArmor
+  ArcAutoProvisioningRoleName:
+    Type: String
+    Description: Provide a role ARN name DefenderForCloud-ArcAutoProvisioning for
+      needed offerings
+    AllowedPattern: '[-_a-zA-Z0-9]+'
+    Default: DefenderForCloud-ArcAutoProvisioning-stackArmor
+  SetupType:
+    Type: String
+    AllowedValues:
+      - Local
+      - Organizational
+    Default: Organizational
+    Description: '(Required) Specifies the type of the Setup: either local or organizational.'
+  ConfigurationID:
+    Type: String
+    Default: MDCSetup
+    Description: (Required) Unique identifier of the deployed configuration.
+  IsPolicyAttachAllowed:
+    Type: String
+    Default: 'true'
+    AllowedValues:
+      - 'true'
+      - 'false'
+    Description: (Optional) Whether MDc setup is allowed to attach policies to existing
+      Instance profiles
+  ProvidedInstanceProfileName:
+    Type: String
+    Default: ''
+    Description: (Optional) Specifies the instance profile Name provided by the user
+      when SetupType=Local.
+  ProvidedAssumeRoleArn:
+    Type: String
+    Default: '*'
+    Description: (Optional) Specifies the automation assume role Arn provided by the
+      user when SetupType=Local.
+  TargetType:
+    Type: String
+    Default: '*'
+    AllowedValues:
+      - Tags
+      - InstanceIds
+      - '*'
+      - ResourceGroups
+    Description: (Optional) Specifies the way in which instances are targeted - applies
+      only for local MDCSetup.
+  TargetInstances:
+    Type: String
+    Default: '*'
+    Description: (Optional) Specifies the instances to be targeted when SetupType=Local
+      and TargetType=InstanceIds.
+  TargetTagKey:
+    Type: String
+    Default: ''
+    Description: (Optional) Specifies the tag key of instances to be targeted when
+      SetupType=Local and TargetType=Tags
+  TargetTagValue:
+    Type: String
+    Default: ''
+    Description: (Optional) Specifies the tag value of instances to be targeted when
+      SetupType=Local and TargetType=Tags
+  ResourceGroupName:
+    Type: String
+    Default: ''
+    Description: (Optional) Specifies the resource group name to be targeted when
+      SetupType=Local and TargetType=ResourceGroups
+  DataSecurityPostureDbRoleName:
+    Type: String
+    Description: Provide a role ARN name DefenderForCloud-DataSecurityPostureDB for
+      Databases DSPM offering
+    AllowedPattern: '[-_a-zA-Z0-9]+'
+    Default: DefenderForCloud-DataSecurityPostureDB-stackArmor
+  IsManagementAccountOnboarding:
+    Type: String
+    Default: 'true'
+    AllowedValues:
+      - 'true'
+      - 'false'
+    Description: (Optional) Whether MDC setup is management account onboarding
+  OidcAccountId:
+    Type: String
+    Default: '505541372182'
```

Comment:
An attempt to deploy failed with an error that indicates that AWS cannot find `DefenderForCloud-OidcCiem-stackArmor` role in  AWS account `505541372182`.  I'm assuming that this is a StackArmor account @frsfx ? ...

```
Resource handler returned message: "Invalid principal in policy:
"AWS":"arn:aws:iam::505541372182:role/DefenderForCloud-OidcCiem-stackArmor"
(Service: Iam, Status Code: 400, Request ID: 4329ee5c-8e37-462d-9ec7-2fb18f62d53b)"
(RequestToken: ec588a8a-fad2-f77c-9680-4e89bf402f60, HandlerErrorCode: InvalidRequest)
```

Reply:
That needs to be removed since we are running the template in one account.  I'll update the template

Reply:
I made a suggested change to the template.  Please review and incorporate.